tools/distpartitioning/convert_partition.py

```diff
@@ -18,7 +18,129 @@
     RESERVED_FIELD_DTYPE,
 )
 from pyarrow import csv
-from utils import get_idranges, memory_snapshot, read_json
+from tools.distpartitioning.utils import (
+    get_idranges,
+    memory_snapshot,
+    read_json,
+)
+
+
+def _get_unique_invidx(srcids, dstids, nids):
+    """This function is used to compute a list of unique elements,
+    and their indices in the input list, which is the concatenation
+    of srcids, dstids and uniq_nids. In addition, this function will also
+    compute inverse indices for the elements in srcids and dstids arrays.
+    In the process srcids, dstids will be over-written to contain the
+    inverse indices, which will the indices in the unique elements list.
+    Basically, this function is mimicing the functionality of numpy's
+    unique function call. The problem with numpy's unique function call
+    is its high memory requirement. For an input list of 3 billion edges
+    it consumes about 550GB of systems memory, which is limiting the
+    capability of the partitioning pipeline.
+
+    Parameters:
+    -----------
+    srcids : numpy array
+        a list of numbers, and in our use-case this will be src-ids of
+        the edges
+    dstids : numpy array
+        a list of numbers, and in our use-case this will be dst-ids of
+        the edges
+    nids : numpy array
+        a list of numbers, and in our use-case this will be a list of
+        unique shuffle-global-nids on a given rank
+
+    Returns:
+    --------
+    numpy array :
+        a list of unique, sorted elements, computed from the input arguments
+    numpy array :
+        a list of integers. These are indices in the concatenated list
+        [srcids, dstids, uniq_nids], which are the input arguments to this function
+    numpy array :
+        a list of integers. These are inverse indices, which will be indices
+        from the unique elements list specifying the elements from the
+        input array, srcids
+    numpy array :
+        a list of integers. These are inverse indices, which will be indices
+        from the unique elements list specifying the elements from the
+        input array, dstids
+
+    """
+    mask = np.in1d(srcids, nids, invert=True, kind="table")
```

Comment:
`kind=` requires `numpy >=1.24` which not available in CI.

Reply:
Testing the new functionality without kind.